airflow/contrib/operators/gcp_function_operator.py

```diff
@@ -0,0 +1,571 @@
+        :param dictionary_to_validate: dictionary that must follow the specification
+        :return: None
+        """
+        try:
+            for validation_spec in self._validation_specs:
+                self._validate_field(validation_spec=validation_spec,
+                                     dictionary_to_validate=dictionary_to_validate)
+        except FieldValidationException as e:
+            raise FieldValidationException(
+                "There was an error when validating: '{}': {}".
+                format(dictionary_to_validate, e))
+
+# TODO End of field validator to be extracted
+
+
+def _validate_available_memory_in_mb(value):
+    if int(value) <= 0:
+        raise FieldValidationException("The available memory has to be greater than 0")
+
+
+def _validate_max_instances(value):
+    if int(value) <= 0:
+        raise FieldValidationException(
+            "The max instances parameter has to be greater than 0")
+
+
+LABEL_KEY_REGEXP = r'^[\w_-]{0,62}$'
+LABEL_VALUE_REGEXP = r'^[\w_-]{0,63}$'
+
+
+def _validate_labels_dict(the_dictionary):
+    # type: (dict) -> None
+    if not isinstance(the_dictionary, dict):
+        raise FieldValidationException("The field should be dictionary!")
+    for key in the_dictionary.keys():
+        if not re.match(LABEL_KEY_REGEXP, key):
+            raise FieldValidationException("The key '{}' does not match regexp '{}'".
+                                           format(key, LABEL_KEY_REGEXP))
+        value = the_dictionary[key]
+        if not re.match(LABEL_VALUE_REGEXP, value):
+            raise FieldValidationException(
+                "The value '{}' of key '{}' does not match regexp '{}'".
+                format(value, key, LABEL_VALUE_REGEXP))
+
+
+ENV_VARS_KEY_REGEXP = r'^[^\x00=]+$'
+ENV_VARS_VALUE_REGEXP = r'^[^\\x00]*$'
+
+
+def _validate_env_vars_dict(the_dictionary):
+    # type: (dict) -> None
+    if not isinstance(the_dictionary, dict):
+        raise FieldValidationException("The field should be dictionary!")
+    for key in the_dictionary.keys():
+        if key is None:
+            raise FieldValidationException("The key '{}' cannot be None".format(key))
+        if not re.match(ENV_VARS_KEY_REGEXP, key):
+            raise FieldValidationException("The key '{}' does not match regexp '{}'".
+                                           format(key, ENV_VARS_KEY_REGEXP))
+        value = the_dictionary[key]
+        if not re.match(ENV_VARS_VALUE_REGEXP, value):
+            raise FieldValidationException(
+                "The value '{}' of key '{}' does not match regexp '{}'".
+                format(value, key, ENV_VARS_VALUE_REGEXP))
+
+
+CLOUD_FUNCTION_VALIDATION = [
+    dict(name="name", regexp=r'^[^\s]+$'),
+    dict(name="description", regexp="^.+$", optional=True),
+    dict(name="entryPoint", optional=True, regexp=r'^\w+$'),
+    dict(name="runtime", regexp=r'^nodejs6$|^nodejs8$|^python37$', optional=True),
```

Comment:
+1

Reply:
Agree. From this and further comments I understand better what "non-breaking change" might mean and I fully agree those validations might be harmful. I will make a small change to enable validator to just validate presence of fields, so that we can remove regexp altogether from pretty much all the fields.

Reply:
Fixed in cb0bba61097c9c6a45538392015a4394f70e8905